### ISSUE TITLE
Renew topic subscriptions

### ISSUE THREAD

**Issue by chris13524** [WalletConnect/notify-server#325]
Subscriptions expire after 30 days, we need to renew them periodically

**Comment by chris13524** [comment]
Topic subscriptions are technically ephemeral until persistence is implemented in the relay. See [discussion](https://www.notion.so/Permissioned-Network-Launch-Checklist-591a8ab0754448388e310b700da7c168?d=291937f27c244c1aa3f09519c9d2741c&pvs=4#1ab5effcf5c44a7e86a9521900f40eeb).

**Comment by chris13524** [comment]
Ideally we have a separate ECS task that does this independently. And this could be horizontally scaled if necessary in the future. But needing horizontal scaling here seems silly and speaks to larger architectural issues. And splitting this into a separate task/service will be a bit of work and will make things more complicated in the future.

The lock might be the next best option, but takes some effort to implement and could be a failure point.

I think the best option for now is just to implement the renewal logic, but then add significantly random offsets to the run interval. E.g. check for renewal daily, but +/- 12 hours. Renewing all 600k topics currently takes a couple of hours, and so renewing only the ones required (at 600k currently) only needs 1 topic to be renewed every 4 seconds. However these requests would be batched and so would rarely duplicate. Duplicate requests even at this frequency are not a significant concern to the health of the relay.

One more thing is that initially all the topics will be renewed at the same time, which would still result in spikes of renewals. It would be desirable to add an initial random spread to the renewal times over the course of 30 days, as well as adding randomness to the renewal date itself to keep the spread random over-time.

**Comment by heilhead** [comment]
> publishing a 4050 message to the topic in order to extend the TTL to 30 days

The legacy storage implementation [requires at least two messages](https://github.com/WalletConnect/rs-relay/blob/f06342c90d16042a75238d418ff69b6544d942f1/src/storage/routing.rs#L278) to extend TTL.

**Comment by chris13524** [comment]
> > publishing a 4050 message to the topic in order to extend the TTL to 30 days
>
> The legacy storage implementation [requires at least two messages](https://github.com/WalletConnect/rs-relay/blob/f06342c90d16042a75238d418ff69b6544d942f1/src/storage/routing.rs#L278) to extend TTL.

Or a message in the 4xxx range, which is what we are doing here

**Comment by heilhead** [comment]
> > > publishing a 4050 message to the topic in order to extend the TTL to 30 days
> > 
> > 
> > The legacy storage implementation [requires at least two messages](https://github.com/WalletConnect/rs-relay/blob/f06342c90d16042a75238d418ff69b6544d942f1/src/storage/routing.rs#L278) to extend TTL.
> 
> Or a message in the 4xxx range, which is what we are doing here

Oh yeah, the code was modified since I implemented it. My bad.

**Comment by chris13524** [comment]
Since we switched to IRN, we no longer need to publish a message to extend the subscription TTL to 30 days. Now the subscription TTL [is 30 days by default](https://walletconnect.slack.com/archives/C03SMNKLPU0/p1708448189986729?thread_ts=1708447036.412879&cid=C03SMNKLPU0). We may revert so let's not depend on this behavior for next couple weeks.

This changes the math and potentially the strategy (do we need to touch the database now?). Subscribing to all topics historically only took a couple of minutes when running in parallel.

- [Can we use batchSubscribe?](https://walletconnect.slack.com/archives/C03SMNKLPU0/p1708447036412879)

**Comment by chris13524** [comment]
For now, renewing all topic subscriptions with cheap method: https://github.com/WalletConnect/notify-server/pull/388

In the future we can renew topic subscriptions only when the need to be.
